### ISSUE TITLE
Add a type for currency

### ISSUE THREAD

**Issue by dsherry** [alteryx/woodwork#769]
Currently woodwork will detect data of the form
```
$1.234
$5.678
...
```
as "Natural Language". It would be helpful if we created a currency type so that this sort of data gets picked up as numeric by default.

**Comment by BoopBoopBeepBoop** [comment]
Which currencies? Currency math typically operates using decimals. Would these types use fixed point storage?

**Comment by thehomebrewnerd** [comment]
In addition to the decimal concern raised by @BoopBoopBeepBoop , we would also need to think about the different global currency symbols used as well as different usage of commas and decimal points around the world unless we wanted this to be a US-only inference.

**Comment by dsherry** [comment]
@BoopBoopBeepBoop yeah good questions! I think the physical type for logical type "Currency" should be a float.

Could something like [this library](http://babel.pocoo.org/en/latest/api/numbers.html) help with parsing and with internationalization?

I do think there's value to be had in starting with one localization (e.g. USD), to get the feature working, and to then add support for other localizations.

It occurs to me, it would be cool if woodwork tagged currency columns with the name of the currency they use!

**Comment by gsheni** [comment]
If a user passes a column with datetime values `[12/12/12, 1/1/1, 2/2/2]`, we will infer/change theses values to be a datetime and add Hour, Minute, Second. 
So passing a column with currency symbols, and changing to a Float not a radically different behavior.

**Comment by willsmithorg** [comment]
In the finance industry its rare to have a numerical value and currency stored in a single feature.

Instead normally you'd have something like
Dividend_currency. (iso4217 3 letter code)
Dividend_amount (numeric)

There would be value in 
1) creating a logical feature to store and detect currency codes (iso4217 symbols)
2) detecting column pairs like the above and automatically converting to USD using a recent exchange rate. That would permit rows in different currencies to be compared more fairly based on value.  For example, 14000 IDR is around 1 USD is unless we convert

**Comment by gsheni** [comment]
@willsmithorg You would have to tag the numeric column, perhaps with a `currency` tag to specify that its currency values. This semantic tag (**currency**) could work with Double OR Integer Logical Type.

**Comment by gsheni** [comment]
Another way to tackle this problem is to turn currency into a tuple of values, similar to LatLong. Each value could then have a corresponding currency symbol/code.

**Comment by thehomebrewnerd** [comment]
A related article that might be useful when working on a solution for this: https://cs-syd.eu/posts/2022-08-22-how-to-deal-with-money-in-software
